atomicfu/src/jsAndWasmSharedMain/kotlin/kotlinx/atomicfu/locks/Mutex.kt

```diff
@@ -0,0 +1,9 @@
+package kotlinx.atomicfu.locks
+
+actual class Mutex {
+    private var locked = false
+    actual fun isLocked(): Boolean = locked
+    actual fun tryLock(): Boolean = true
+    actual fun lock(): Unit { locked = true }
+    actual fun unlock(): Unit { locked = false }
```

Comment:
Since the mutex is reentrant, you need to maintain a counter instead so the deepest `unlock` call does not release the mutex.